lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsField.js

```diff
@@ -66,13 +70,55 @@ const OptionsField = ({
   };
 
   const getContentData = fields => {
+    if (isMultiSelect) {
+      return fields.map((_field, index) => ({ index }));
+    }
+
     return [{}].concat(fields.map((_field, index) => ({ index: index + 1 })));
   };
 
+  const getLabelField = rowData => {
+    if (!isMultiSelect && !rowData.rowIndex) {
+      return null;
+    }
+
+    const index = (isMultiSelect ? rowData.rowIndex : rowData.rowIndex - 1);
+
+    return (
+      <Field
+        className={css.optionInput}
+        name={`${fieldNamePrefix}.selectField.options.values[${index}]`}
+        validate={validateLabel}
+        validateFields={[]}
+        marginBottom0
+      >
+        {(fieldProps) => (
+          <TextField
+            {...fieldProps}
+            onChange={e => {
+              if (isMultiSelect && defaultOptions.includes(fieldProps.input.value)) {
+                changeFieldValue(`${fieldNamePrefix}.selectField.defaults`, [...defaultOptions, e.target.value]);
```

Comment:
Shouldn't we change the option at `rowData.rowIndex` position specifically?

Reply:
No, the default's array is not connected in any way with option fields and their indexes aren't equal